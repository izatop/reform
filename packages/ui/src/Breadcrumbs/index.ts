export * from "./BreadcrumbsProvider";
export * from "./BreadcrumbsState";
export * from "./Breadcrumbs";
export * from "./functions";
