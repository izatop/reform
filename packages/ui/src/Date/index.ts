export * from "./Calendar";
export * from "./DateRangePicker";
export * from "./functions";
export * from "./interfaces";
