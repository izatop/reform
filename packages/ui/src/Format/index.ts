export * from "./DateFormat";
export * from "./DateTimeFormat";
export * from "./TimeFormat";
