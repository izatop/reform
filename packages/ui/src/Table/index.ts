export * from "./TableConfig";
export * from "./TableFactory";
export * from "./TableProperty";
export * from "./TableRenderer";
export * from "./TableStore";
