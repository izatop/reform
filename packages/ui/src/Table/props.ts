import * as React from "react";
import {TableStore} from "./TableStore";

export const TableStoreContext = React.createContext(new TableStore<any>([]));

export interface ITableSource {
    [key: string]: any;
}

export function isValidElement<P>(child: any, type: React.ElementType<P>): child is React.ReactElement<P> {
    if (!React.isValidElement(child)) {
        return false;
    }

    return child.type === type;
}
