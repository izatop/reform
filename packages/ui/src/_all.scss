@import "~@reform/bulma/sass/variables";
@import "./Date/all";
