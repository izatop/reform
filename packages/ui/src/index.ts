export * from "./Accesibility";
export * from "./Table";
