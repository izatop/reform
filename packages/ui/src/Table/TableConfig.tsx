import * as React from "react";
import {ITableRow, XProps} from "@reform/bulma";
import {ITableSource} from "./props";
import {ITableProperty} from "./TableProperty";

export interface ITableConfig<T extends ITableSource> {
    primary: keyof T & string;
    rowProps?: (data: T) => ITableRow & XProps<"tr">;
    children: React.ReactElement<ITableProperty<T>> | React.ReactElement<ITableProperty<T>>[];
}

export const TableConfig: React.FC<ITableConfig<any>> = () => null;
