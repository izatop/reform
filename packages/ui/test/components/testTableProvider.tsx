import * as React from "react";
import * as renderer from "react-test-renderer";
import {TableFactory} from "../../src/Table";

test("TableFactory", () => {
    const data = [
        {id: 1, f1: 76, f2: "Text 1"},
        {id: 2, f1: 98, f2: "Text 2"},
    ];

    const Table = TableFactory.create(data);

    const element = (
        <Table.Factory data={data}>
            <Table.Config primary={"id"} rowProps={({id}) => (id === 2 ? {"is-selected": true} : {})}>
                <Table.Property name={"f1"} align={"right"}/>
                <Table.Property name={"f2"} total={() => "total"}>
                    {({id, f2}) => (<span>Hello, {f2} ({id})</span>)}
                </Table.Property>
            </Table.Config>
            <Table.Renderer footer/>
        </Table.Factory>
    );

    expect(renderer.create(element).toJSON())
        .toMatchSnapshot();
});
