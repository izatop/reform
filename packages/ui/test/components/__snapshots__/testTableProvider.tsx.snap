// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableFactory 1`] = `
<table
  className="table"
>
  <thead>
    <tr
      className=""
    >
      <th
        className=""
      />
      <th
        className=""
      />
    </tr>
  </thead>
  <tfoot>
    <tr
      className=""
    >
      <th
        className=""
      />
      <th
        className=""
      >
        total
      </th>
    </tr>
  </tfoot>
  <tbody>
    <tr
      className=""
    >
      <td
        className=""
      >
        76
      </td>
      <td
        className=""
      >
        <span>
          Hello, 
          Text 1
           (
          1
          )
        </span>
      </td>
    </tr>
    <tr
      className=""
    >
      <td
        className=""
      >
        98
      </td>
      <td
        className=""
      >
        <span>
          Hello, 
          Text 2
           (
          2
          )
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
